plugin-server/src/cdp/utils.ts

```diff
@@ -378,3 +379,66 @@ export function cyclotronJobToInvocation(job: CyclotronJob, hogFunction: HogFunc
         timings: parsedState.timings,
     }
 }
+
+const CYCLOTRON_GAUGES: Map<string, Gauge> = new Map()
+const CYCLOTRON_COUNTERS: Map<string, Counter> = new Map()
+const CYCLOTRON_HISTOGRAMS: Map<string, Histogram> = new Map()
+const CYCLOTRON_HISTOGRAM_BOUNDS: number[] = exponentialBuckets(0.5, 2, 11) // 0.5 -> 512
+
+export function emitCyclotronMetrics() {
+    const report = getMetricsReport()
+    for (const measurement of report.measurements) {
+        const name = measurement.name
+        const type = measurement.type
+
+        // The TS library seem to demand you declare labels up-front (for some reason?),
+        // but we have know good way of knowing the set of labels used up-front, so
+        // idk what to do here really - recreate the metric each time? That seems...
```

Comment:
this would need to be tested but it could be that the labels are just for typing so if you just cast the label setting `as any` it might JustWork. Checked the lib source and it wasn't obvious either way if it would work though...

Reply:
Found the ever-classic 5 year old issue 🙃 https://github.com/siimon/prom-client/issues/298